pwndbg/elf.py

```diff
@@ -68,6 +70,59 @@ def read(typ, address, blob=None):
     return obj
 
 
+@pwndbg.proc.OnlyWhenRunning
+def get_containing_segments(elf_filename, vaddr, loadaddr):
+    local_path = pwndbg.file.get_file(elf_filename)
+    segments = []
+    with open(local_path, 'rb') as f:
+        elffile = ELFFile(f)
+        is_pic = elffile.header['e_type'] == 'ET_DYN'
```

Comment:
Maybe we should expose `pwndbg.elf.is_pic`?

Reply:
It could make sense to exposed a cached view of ELF properties. However, I think it could be a bit ambiguous maybe to put it as a global property on pwndbg.elf. I realise that usually you want to know whether the main binary is PIC or not (since libraries will today always be PIC). However, one of the bugs in the original xinfo fell into this trap seeing as it always opened the main executable in readelf when finding ELF segments. However, xinfo can of course be applied to something other than the main executable, so that sometimes produced an exception.

Pushing the logic into a separate memoized function in pwndbg.elf could make a lot of sense though. I can craft another commit implementing this